glib/src/gobject/binding_group.rs

```diff
@@ -233,8 +249,197 @@ impl<'a> BindingGroupBuilder<'a> {
     }
 
     // rustdoc-stripper-ignore-next
-    /// Similar to `try_build` but fails instead of panicking.
+    /// Similar to `try_build` but panics instead of failing.
     pub fn build(self) {
         self.try_build().unwrap()
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::prelude::*;
+    use crate::subclass::prelude::*;
+
+    #[test]
+    fn binding_without_source() {
+        let binding_group = crate::BindingGroup::new();
+
+        let source = TestObject::default();
+        let target = TestObject::default();
+
+        assert!(source.find_property("name").is_some());
+        binding_group
+            .bind("name", &target, "name")
+            .flags(crate::BindingFlags::BIDIRECTIONAL)
+            .build();
+
+        binding_group.set_source(Some(&source));
+
+        source.set_name("test_source_name");
+        assert_eq!(source.name(), target.name());
+
+        target.set_name("test_target_name");
+        assert_eq!(source.name(), target.name());
+    }
+
+    #[test]
+    fn binding_with_source() {
+        let binding_group = crate::BindingGroup::new();
+
+        let source = TestObject::default();
+        let target = TestObject::default();
+
+        binding_group.set_source(Some(&source));
+
+        binding_group.bind("name", &target, "name").build();
+
+        source.set_name("test_source_name");
+        assert_eq!(source.name(), target.name());
+    }
+
+    #[test]
+    fn binding_to_transform() {
+        let binding_group = crate::BindingGroup::new();
+
+        let source = TestObject::default();
+        let target = TestObject::default();
+
+        binding_group.set_source(Some(&source));
+        binding_group
+            .bind("name", &target, "name")
+            .flags(crate::BindingFlags::SYNC_CREATE)
+            .transform_to(|_binding, value| {
+                let value = value.get::<&str>().unwrap();
+                Some(format!("{} World", value).to_value())
+            })
+            .build();
+
+        source.set_name("Hello");
+        assert_eq!(target.name(), "Hello World");
+    }
+
+    #[test]
+    fn binding_from_transform() {
```

Comment:
Would be good to have transform tests where the source/target property have different types.

Reply:
give me an example of different types and ill code it, I tried using the current 2 properties (bool and string) and of course it was not possible due to the checks we do, even if we can do the conversion in the GtkExpression.

Reply:
See https://github.com/gtk-rs/gtk-rs-core/issues/700